pkg/storage/s3/checker.go

```diff
@@ -19,27 +19,31 @@ func (s *Storage) Exists(ctx context.Context, module, version string) (bool, err
 	ctx, span := observ.StartSpan(ctx, op.String())
 	defer span.End()
 
-	lsParams := &s3.ListObjectsInput{
-		Bucket: aws.String(s.bucket),
-		Prefix: aws.String(fmt.Sprintf("%s/@v", module)),
+	files := []string{"info", "mod", "zip"}
+	errChan := make(chan error, len(files))
+	defer close(errChan)
+	cancelingCtx, cancel := context.WithCancel(ctx)
```

Comment:
I think this should go in the go routine so that each remote cal has its own ctx

Reply:
Having all goroutine sharing the same `cancelingCtx` allows one failing goroutine to cancel other calls to S3: if we already know one of the file don't exist, we can cancel other `HeadObject` calls.